x-pack/plugins/canvas/canvas_plugin_src/uis/arguments/datacolumn/index.js

```diff
@@ -39,7 +39,7 @@ const getMathValue = (argValue, columns) => {
 // TODO: Garbage, we could make a much nicer math form that can handle way more.
 const DatacolumnArgInput = ({
   onValueChange,
-  columns,
+  resolved: { columns },
```

Comment:
Can you explain the chain that led to some of these args having input come in via this `resolved` instead of directly as it previously was?

Reply:
Sure. Before,  at the `Arg` class we were creating templateProps and passing there everything, but required only a few of the props, which were passed. So, I've reduced the number of props, we're passing and the performance has raised.
https://github.com/elastic/kibana/blob/1abcf8138c64837d62573d3e9afdc6377b5f41dc/x-pack/plugins/canvas/public/expression_types/arg.ts#L142
 
But also, we have to control things, which are passed to the arguments.
At the previous version of this code you could see such logic: 
```ts
const templateProps = { ...otherProps, ...resolved, onValueChange, typeInstance: this };
```
, where `resolved` are the props, which are produced by `models/views/transforms` `resolve` method. Example: https://github.com/elastic/kibana/blob/1abcf8138c64837d62573d3e9afdc6377b5f41dc/x-pack/plugins/canvas/canvas_plugin_src/uis/models/heatmap_grid.ts#L50

Such logic can cause some problems because it is impossible to track if some prop is rewritten. Also, it is difficult to document the types.

At this piece of code (https://github.com/elastic/kibana/blob/1abcf8138c64837d62573d3e9afdc6377b5f41dc/x-pack/plugins/canvas/public/expression_types/arg.ts#L148) you can see, that I've localized the object, produced by `resolve` method and now it is easier to track and understand the way, how our code is really working, and which props are default, and which are specific for our view/model/etc.